packages/next/build/webpack-config.js

```diff
@@ -291,10 +291,18 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer
       dev && new CaseSensitivePathPlugin(), // Since on macOS the filesystem is case-insensitive this will make sure your path are case-sensitive
       !dev && new webpack.HashedModuleIdsPlugin(),
       // Removes server/client code by minifier
-      new webpack.DefinePlugin({
-        'process.crossOrigin': JSON.stringify(config.crossOrigin),
-        'process.browser': JSON.stringify(!isServer)
-      }),
+      new webpack.DefinePlugin(Object.assign(
+        {},
+        config.env ? Object.keys(config.env)
+          .reduce((acc, key) => ({
+            ...acc,
+            ...{ [`process.${key}`]: JSON.stringify(config.env[key]) }
```

Comment:
Shouldn't this be `process.env.${key}` 🤔

Reply:
We should use `process.env.` because otherwise Typescript / Flow etc will not recognize it as a valid format.